frontend/src/app/components/transactions-list/transactions-list.component.html

```diff
@@ -384,15 +384,58 @@
                     <tbody>
                       <tr>
                         <td i18n="transactions-list.scriptpubkey.asm|ScriptPubKey (ASM)">ScriptPubKey (ASM)</td>
-                        <td style="text-align: left;" [innerHTML]="vout.scriptpubkey_asm | asmStyler"></td>
+                        <td style="text-align: left;">
+                          <app-asm [asm]="vout.scriptpubkey_asm" [crop]="showFullScriptPubkeyAsm[vindex] ? 0 : 1000"></app-asm>
+                          <div *ngIf="vout.scriptpubkey_asm.length > 1000" style="display: flex;">
+                            <span *ngIf="!showFullScriptPubkeyAsm[vindex]">...</span>
+                            <label class="btn btn-sm btn-primary mt-2" (click)="toggleShowFullScriptPubkeyAsm(vindex)" style="margin-left: auto;">
+                              <span *ngIf="!showFullScriptPubkeyAsm[vindex]" i18n="show-all">Show all</span>
+                              <span *ngIf="showFullScriptPubkeyAsm[vindex]" i18n="show-less">Show less</span>
+                            </label>
+                          </div>
+                        </td>
                       </tr>
                       <tr>
                         <td i18n="transactions-list.scriptpubkey.hex|ScriptPubKey (HEX)">ScriptPubKey (HEX)</td>
-                        <td style="text-align: left;">{{ vout.scriptpubkey }}</td>
+                        <td style="text-align: left;">
+                          @if (vout.scriptpubkey.length > 1000) {
+                            @if (!showFullScriptPubkeyHex[vindex]) {
+                              {{ vout.scriptpubkey | slice:0:1000 }}
+                            } @else {
+                              {{ vout.scriptpubkey }}
+                            }
+                          } @else {
+                            {{ vout.scriptpubkey }}
+                          }
```

Comment:
the outer `@if ... @else` here is redundant (if the scriptpubkey is less than 1000 characters, the `slice` pipe will have no effect anyway)